### ISSUE TITLE
Possible bug in read.genepop: 123546 read as 12 34 56 and not 123 456

### ISSUE THREAD

**Issue by thibautjombart** [thibautjombart/adegenet#107]
William Engels [wrengels@wisc.edu]

11 November 2015 21:54
Dear Dr. Jombart,

    I am writing to report an apparent bug in adegenet 2 when importing genepop files. I'll enclose a short sample genepop file which I downloaded from a publication. This file was imported correctly with the previous adegenet version, and the commands:
```r
u <- import2genind("urchins.en")
genind2df(u, sep="/")
```
produced a data frame showing the diploid genotypes properly. However, the same commands with v. 2 yields a table showing triploid genotypes as follows:
```r
   pop      C10     C101
1   42 25/22/55 10/91/21
2   42 25/22/55 21/12/11
3   42 25/22/52 10/91/21
4   42 52/24/92 21/12/11
5   42 25/55/52 10/91/21
6   42 25/55/52 12/11/33
7   42 55/24/92 21/12/11
8   42 25/22/55 12/11/33
9   42 25/55/52 10/91/21
10  42 25/52/58 21/12/11
...
```
So the genotypes are being interpreted as three alleles with two characters each instead of two alleles with three characters each.

Best wishes,

Bill Engels

**Comment by thibautjombart** [comment]
[urchins.gen.txt](https://github.com/thibautjombart/adegenet/files/36584/urchins.gen.txt)

**Comment by thibautjombart** [comment]
It wasn't a bug, but you need to use `ncode=3` in `read.genepop`. This argument can now also be passed through `...` to `import2genind`, as of:

ad4fc7df811c59ed954d8a8130afd6a11940fe9f

**Comment by thibautjombart** [comment]
Also note, `read.genepop` will now check for this and issue a meaningful error message if needed:
```r
> import2genind("urchins.gen")

 Converting data from a Genepop .gen file to a genind object... 


File description:  P. bajulus trinucleotides 
Error in read.genepop(file, quiet = quiet, ...) (from import.R#621) : 
  some data are not encoded with 2 characters
Check 'ncode' argument
> import2genind("urchins.gen",ncode=3)

 Converting data from a Genepop .gen file to a genind object... 


File description:  P. bajulus trinucleotides 

...done.

/// GENIND OBJECT /////////

 // 25 individuals; 2 loci; 7 alleles; size: 7.8 Kb

 // Basic content
   @tab:  25 x 7 matrix of allele counts
   @loc.n.all: number of alleles per locus (range: 3-4)
   @loc.fac: locus factor for the 7 columns of @tab
   @all.names: list of allele names for each locus
   @ploidy: ploidy of each individual  (range: 2-2)
   @type:  codom
   @call: read.genepop(file = file, ncode = 3, quiet = quiet)

 // Optional content
   @pop: population of each individual (group size range: 10-15)
> genind2df(import2genind("urchins.gen",ncode=3),sep="/")

 Converting data from a Genepop .gen file to a genind object... 


File description:  P. bajulus trinucleotides 

...done.

   pop     C10    C101
1   42 252/255 109/121
2   42 252/255 121/121
3   42 252/252 109/121
4   42 252/249 121/121
5   42 255/255 109/121
6   42 255/255 121/133
7   42 255/249 121/121
8   42 252/255 121/133
9   42 255/255 109/121
10  42 255/258 121/121
11  42 249/249 121/133
12  42 255/255 121/133
13  42    <NA> 121/121
14  42 255/255 121/121
15  42 255/249 109/121
16  40 255/249 121/121
17  40 252/255 109/121
18  40 255/249 121/133
19  40 249/249 121/133
20  40 249/249 121/121
21  40 249/249 121/133
22  40 252/255 121/133
23  40 252/249 121/121
24  40 255/249 121/133
25  40 255/249 121/121

```

**Comment by thibautjombart** [comment]
Fixed typo in error message at d4f80b612b18f5f096a2c4c0b19093a2368372db
